src/Covid19Dashboard/IllinoisMapChart/index.jsx

```diff
@@ -67,6 +67,52 @@ class IllinoisMapChart extends React.Component {
       },
       hoverInfo: null,
     };
+    this.mapData = {
+      modeledCountyGeoJson: null,
+      colors: {},
+      colorsAsList: null,
+    };
+  }
+  componentDidUpdate() {
+    if (!(this.mapData.colorsAsList === null
+          && Object.keys(this.props.jsonByLevel.county).length > 0)) {
+      return;
+    }
+    if (Object.keys(this.props.jsonByLevel.country).length && !this.choroCountyGeoJson) {
+      this.choroCountyGeoJson = addDataToGeoJsonBase(
+        this.props.jsonByLevel.county,
+      );
+    }
+    this.mapData.modeledCountyGeoJson = filterCountyGeoJson(this.props.modeledFipsList);
+
+    // Finds max value in data set then finds its base 10 exponent
+    const maxValExponent = Math.log10(this.mapData.modeledCountyGeoJson.features
+      .map(obj => obj.properties.confirmed)
+      .sort((a, b) => b - a)[0]);
```

Comment:
can we use the `max` function instead?

Reply:
sort vs max speed test https://jsben.ch/hoZGp looks to depend on the browser, chrome has optimized max